presto-cli/src/main/java/com/facebook/presto/cli/AbstractWarningsPrinter.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cli;
+
+import com.facebook.presto.spi.PrestoWarning;
+
+import java.util.List;
+import java.util.OptionalInt;
+
+import static com.facebook.presto.cli.ConsolePrinter.REAL_TERMINAL;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+abstract class AbstractWarningsPrinter
+        implements WarningsPrinter
+{
+    private static final String WARNING_BEGIN = ((char) 27) + "[33m";
+    private static final String WARNING_END = ((char) 27) + "[39m";
+
+    private final OptionalInt maxWarnings;
+    private boolean hasProcessedWarnings;
+    private int processedWarnings;
+
+    AbstractWarningsPrinter(OptionalInt maxWarnings)
+    {
+        this.maxWarnings = requireNonNull(maxWarnings, "maxWarnings is null");
+    }
+
+    private String getWarningMessage(PrestoWarning warning)
+    {
+        // If this is a real terminal color the warnings yellow
+        if (REAL_TERMINAL) {
+            return format("%sWARNING: %s%s", WARNING_BEGIN, warning.getMessage(), WARNING_END);
+        }
+        return format("WARNING: %s", warning.getMessage());
+    }
+
+    private List<String> getNewWarnings(List<PrestoWarning> warnings)
+    {
+        int end;
+        if (maxWarnings.isPresent()) {
```

Comment:
This could be simplified to:

```java
int end = Math.min(processedWarnings + maxWarnings.orElse(Integer.MAX_VALUE), warnings.size());
```

Alternatively, you could write it as follows, which makes the intent a little clearer.

```java
int end = warnings.size();
if (maxWarnings.isPresent()) {
    end = Math.min(end, processedWarnings + maxWarnings.getAsInt());
}
```